Source/WebCore/page/Navigator.cpp

```diff
@@ -194,8 +194,12 @@ void Navigator::share(Document& document, const ShareData& data, Ref<DeferredPro
         if (m_loader)
             m_loader->cancel();
 
-        m_loader = ShareDataReader::create([this, promise = WTFMove(promise)](ExceptionOr<ShareDataWithParsedURL&> readData) mutable {
-            showShareData(readData, WTFMove(promise));
+        m_loader = ShareDataReader::create([weakThis = WeakPtr { *this }, promise = WTFMove(promise)](ExceptionOr<ShareDataWithParsedURL&> readData) mutable {
+            RefPtr protectedThis = weakThis.get();
+            if (!protectedThis)
```

Comment:
```suggestion
            if (!protectedThis) {
                return;
            }
```

Great work @atar13, LGTM. A nit: it's good practice to be in the habit of unconditionally putting curlies around conditional statements, especially with control flow. This prevents subtle bugs like Apple's [`goto` vulnerability](https://www.blackduck.com/blog/understanding-apple-goto-fail-vulnerability-2.html) ([longer explanation](https://dwheeler.com/essays/apple-goto-fail.html)).

Reply:
This suggestion goes against WebKit coding style (https://webkit.org/code-style-guidelines/#braces-one-line).

Reply:
Yeah, definitely don't do that in WebKit.